### ISSUE TITLE
`x in range(...)` should be allowed where x is not a number

### ISSUE THREAD

**Issue by brandjon** [bazelbuild/starlark#175]
> The `x in y` operator, where `y` is a range, reports whether `x` is equal to some member of the sequence `y`; the operation fails unless `x` is a number.

This is strange; `x in y` should have the same semantics as `x in list(y)`.

(First reported in bazelbuild/bazel#8722.)

**Comment by ndmitchell** [comment]
Agreed - the Rust Starlark implementation at https://github.com/facebookexperimental/starlark-rust/ returns False in such cases.
